### DISCUSSION TITLE
Esp32 p4

### DISCUSSION THREAD

**Discussion by maaamcube** [cherry-embedded/CherryUSB#266] [General]
Hello everyone,
Can CherryUSB run multi hubs and multi devices on the Esp32 p4? and what is the limit ? for example ; i have to run 3 hubs and 3 devices at the same time then send it through wifi?
I wanted to use the esp32s3 but it can use 8 channels maximum for the host and 16 channels maximum for the esp32p4.
Thanks.

**Comment by sakumisu** [comment]
Not test host in esp32p4, but esp32s3 can do.
